shell-completion/bash/stratis

```diff
@@ -1,70 +1,116 @@
 _stratis()
 {
-	local cur prev x2prev opts root_subcommands pool_subcommands fs_subcommands blockdev_subcommands daemon_subcommands complete_pools blockdevs first second third
+	local comp_cword_adj root_subcommand subcommand subcommand_opt cur prev opts root_subcommands pool_subcommands fs_subcommands blockdev_subcommands daemon_subcommands complete_pools blockdevs
 
 	COMPREPLY=()
 
 	cur="${COMP_WORDS[COMP_CWORD]}"
 	prev="${COMP_WORDS[COMP_CWORD-1]}"
-	x2prev="${COMP_WORDS[COMP_CWORD-2]}"
-	first="${COMP_WORDS[1]}"
-	second="${COMP_WORDS[2]}"
-	third="${COMP_WORDS[3]}"
 	opts="-h --help"
-	global_opts="--version --propagate ${opts}"
-	listing_opts="--unhyphenated-uuids"
+	global_opts="--version --propagate ${opts} --unhyphenated-uuids"
 	root_subcommands="daemon pool blockdev filesystem key report"
 	pool_subcommands="add-cache add-data bind create destroy init-cache list rename unbind unlock"
 	pool_create_opts="--redundancy --key-desc"
 	pool_bind_opts="--thumbprint --trust-url"
 	pool_bind_subcommands="nbde tang tpm2"
 	pool_unlock_subcommands="clevis keyring"
 	fs_subcommands="create snapshot list rename destroy"
+	fs_create_opts="--size"
 	blockdev_subcommands="list"
 	daemon_subcommands="redundancy version"
 	key_subcommands="set unset reset list"
 	key_set_reset_opts="--keyfile-path --capture-key"
 	blockdevs="$(echo /dev/disk/*/*) $(lsblk -n -o name -p -l)"
+	root_subcommand=""
+	subcommand=""
+	subcommand_first_param=""
+	subcommand_opt=""
+	comp_cword_adj=${COMP_CWORD}
+	
+	# find root subcommand and command if possible
+	if [[ ${COMP_CWORD} -ge 2 ]]; then
+		# find the root subcommand and compute an adjusted value of COMP_CWORD
+		# that doesn't count options
+		for el in ${COMP_WORDS[@]}; do
+			if [[ ${el} != -* && ${el} != "none" ]]; then
+				if [[ ${root_subcommand} == "" && ${el} != "stratis" ]]; then
+					root_subcommand=${el}
+				elif [[ ${root_subcommand} != "" && ${subcommand} == "" ]]; then
+					subcommand=${el}
+				elif [[ ${root_subcommand} != "" && ${subcommand} != "" && ${subcommand_first_param} == "" ]]; then
+					subcommand_first_param=${el}
+				fi
+			else
+				if [[ ${root_subcommand} != "" && ${subcommand} != "" && ${subcommand_opt} == "" ]]; then
+					subcommand_opt=${el}
+				fi
+				let "comp_cword_adj--"
+			fi
+
+		done
+		# find the subcommand 
```

Comment:
There is a space at the end of `# find the subcommand`, and there are 4 tabs on the empty line on line 55, just before `break`.